rust/gui-client/src-tauri/src/client/deep_link.rs

```diff
@@ -1,43 +1,60 @@
 //! A module for registering, catching, and parsing deep links that are sent over to the app's already-running instance
-//! Based on reading some of the Windows code from <https://github.com/FabianLars/tauri-plugin-deep-link>, which is licensed "MIT OR Apache-2.0"
 
 use crate::client::auth::Response as AuthResponse;
-use connlib_shared::{control::SecureUrl, BUNDLE_ID};
+use connlib_shared::control::SecureUrl;
 use secrecy::{ExposeSecret, Secret, SecretString};
-use std::{ffi::c_void, io, path::Path, str::FromStr};
-use tokio::{io::AsyncReadExt, io::AsyncWriteExt, net::windows::named_pipe};
-use windows::Win32::Security as WinSec;
+#[cfg(target_os = "windows")]
+use std::io;
 
 pub(crate) const FZ_SCHEME: &str = "firezone-fd0020211111";
 
+#[cfg(target_os = "linux")]
+#[path = "deep_link/linux.rs"]
+mod imp;
+
+#[cfg(target_os = "windows")]
+#[path = "deep_link/windows.rs"]
+mod imp;
+
+// TODO: The repeated `target_os = "windows"` here is bad.
+// When I add deep link support for Linux I'll replace this all for `anyhow`.
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     #[error("named pipe server couldn't start listening, we are probably the second instance")]
     CantListen,
     /// Error from client's POV
+    #[cfg(target_os = "windows")]
```

Comment:
until we replace this with anyhow can't we just have this for both platforms, we can allow dead code for linux for this enum if the compiler complains